File: 0455-assign-cookies/0455-assign-cookies.dart
class Solution {
  int findContentChildren(List<int> g, List<int> s) {
    g.sort();
    s.sort();
    int p1 = 0;
    int p2 = 0;
    int counter = 0;

    while (p1 < g.length && p2 < s.length) {
      if (g[p1] <= s[p2]) {
        counter += 1;
        p1++;
      }
      p2++;
    }

    return counter;
  }
}

File: 0455-assign-cookies/0455-assign-cookies.swift
class Solution {
    func findContentChildren(_ g: [Int], _ s: [Int]) -> Int {
        // Non Mutating => sorted gives a new list
        // Mutating => sort gives the same array but sorted used with let
        let sortedG : [Int] = g.sorted()
        let sortedS : [Int] = s.sorted()
        var counter : Int = 0
        var p1:Int = 0
        var p2:Int = 0
        while p1 < sortedG.count && p2 < sortedS.count {
            if sortedG[p1] <= sortedS[p2]{
                p1 += 1
                counter += 1
            }
            p2 += 1
        }
        return counter
    }
}